com.unity.ml-agents/Runtime/SideChannels/SideChannelManager.cs

```diff
@@ -120,6 +120,13 @@ internal static byte[] GetSideChannelMessage()
         /// <returns></returns>
         internal static byte[] GetSideChannelMessage(Dictionary<Guid, SideChannel> sideChannels)
         {
+            if (!HasOutgoingMessages(sideChannels))
+            {
+                // Early out so that we don't create the MemoryStream or BinaryWriter.
+                // This is the most common case.
+                return Array.Empty<byte>();
```

Comment:
I didn't try this with null instead of Empty() but 99% the protobuf code will choke on null.